drivers/display/ssd1327.c

```diff
@@ -67,186 +80,256 @@ static inline int ssd1327_write_bus_cmd(const struct device *dev, const uint8_t
 }
+#endif
 
 static int ssd1327_write(const struct device *dev, const uint16_t x, const uint16_t y,
 			 const struct display_buffer_descriptor *desc, const void *buf)
 {
 	const struct ssd1327_config *config = dev->config;
-	struct display_buffer_descriptor mipi_desc;
 	int err;
 	size_t buf_len;
-	uint8_t x_position[] = { x, x + desc->width - 1 };
-	uint8_t y_position[] = { y, y + desc->height - 1 };
+	int32_t pixel_count = desc->width * desc->height;
+	uint8_t x_position[] = {x / 2, (x + desc->width - 1) / 2};
+	uint8_t y_position[] = {y, y + desc->height - 1};
 
 	if (desc->pitch < desc->width) {
 		LOG_ERR("Pitch is smaller than width");
 		return -1;
 	}
-	mipi_desc.pitch = desc->pitch;
 
 	/* Following the datasheet, in the GDDRAM, two segment are split in one register */
 	buf_len = MIN(desc->buf_size, desc->height * desc->width / 2);
 	if (buf == NULL || buf_len == 0U) {
 		LOG_ERR("Display buffer is not available");
 		return -1;
 	}
-	mipi_desc.buf_size = buf_len;
 
 	if (desc->pitch > desc->width) {
 		LOG_ERR("Unsupported mode");
 		return -1;
 	}
 
-	if ((y & 0x7) != 0U) {
+	if ((x & 1) != 0U) {
 		LOG_ERR("Unsupported origin");
 		return -1;
 	}
-	mipi_desc.height = desc->height;
-	mipi_desc.width = desc->width;
 
 	LOG_DBG("x %u, y %u, pitch %u, width %u, height %u, buf_len %u", x, y, desc->pitch,
 		desc->width, desc->height, buf_len);
 
-	err = ssd1327_write_bus_cmd(dev, SSD1327_SET_COLUMN_ADDR, x_position, sizeof(x_position));
+	err = config->write_cmd(dev, SSD1327_SET_COLUMN_ADDR, x_position, sizeof(x_position));
 	if (err) {
 		return err;
 	}
 
-	err = ssd1327_write_bus_cmd(dev, SSD1327_SET_ROW_ADDR, y_position, sizeof(y_position));
+	err = config->write_cmd(dev, SSD1327_SET_ROW_ADDR, y_position, sizeof(y_position));
 	if (err) {
 		return err;
 	}
 
-	err = mipi_dbi_write_display(config->mipi_dev, &config->dbi_config, buf, &mipi_desc,
-				     PIXEL_FORMAT_MONO10);
-	if (err) {
-		return err;
-	}
-	return mipi_dbi_release(config->mipi_dev, &config->dbi_config);
+	return ssd1327_write_pixels_i2c(dev, buf, pixel_count, desc);
 }
 
 static int ssd1327_set_contrast(const struct device *dev, const uint8_t contrast)
 {
-	return ssd1327_write_bus_cmd(dev, SSD1327_SET_CONTRAST_CTRL, &contrast, 1);
+	const struct ssd1327_config *config = dev->config;
+
+	return config->write_cmd(dev, SSD1327_SET_CONTRAST_CTRL, &contrast, 1);
 }
 
-static void ssd1327_get_capabilities(const struct device *dev,
-				     struct display_capabilities *caps)
+static void ssd1327_get_capabilities(const struct device *dev, struct display_capabilities *caps)
 {
 	const struct ssd1327_config *config = dev->config;
 
 	memset(caps, 0, sizeof(struct display_capabilities));
 	caps->x_resolution = config->width;
 	caps->y_resolution = config->height;
-	caps->supported_pixel_formats = PIXEL_FORMAT_MONO10;
-	caps->current_pixel_format = PIXEL_FORMAT_MONO10;
-	caps->screen_info = SCREEN_INFO_MONO_VTILED;
+	caps->supported_pixel_formats = PIXEL_FORMAT_L_8;
```

Comment:
Would it be feasable to have both?

Reply:
If you mean keep it as it was as a alternative, no, the 'monochrome' mode doesnt actually exists, and it was simply not working at all on my display/interface combination.
If you mean adding a conversion, sure, that is possible.